sys/dev/misc/efirt/efidev.c

```diff
@@ -77,9 +77,24 @@ efidev_ioctl(struct dev_ioctl_args *ap)
 	case EFIIOC_GET_TABLE:
 	{
 		struct efi_get_table_ioc *egtioc =
-		    (struct efi_get_table_ioc *)addr;
+			(struct efi_get_table_ioc *)addr;
+		struct efi_esrt_table *esrt = NULL;	
 
-		error = efi_get_table(&egtioc->uuid, &egtioc->ptr);
+		error = efi_get_table(&egtioc->uuid, (void **)&esrt);
+
+		egtioc->table_len = sizeof(*esrt)+ (sizeof(struct efi_esrt_entry_v1) * esrt->fw_resource_count);
+
+		/* Return table lenght to userspace */
+		if(egtioc->ptr == NULL){
+			break;
+		}
+
+		/* Refuse to copy only part of the table */
+		if(egtioc->buf_len < egtioc->table_len){
+			break;
```

Comment:
This should have `error = EINVAL` before `break` to report the error back to user-space.

Reply:
@SergiiDmytruk now should be ok

Reply:
@SergiiDmytruk can you resolve that conversation if everything is ok?

Reply:
Resolved.